### ISSUE TITLE
Disable drag if focus on particular element in node

### ISSUE THREAD

**Issue by baptisteArno** [jsplumb/jsplumb#1074]
Is there a way to disable the drag of a node only if I click on particular elements nested in a node?

Something [like this](https://reactflow.dev/docs/api/node-types/) (See "Prevent dragging & selecting")

**Comment by sporritt** [comment]
You can do this via a `filter` in the `dragOptions` for your instance:

```
jsPlumbBrowserUI.newInstance({
  dragOptions:{
    filter:".someClass, .someOtherClass"
  }
});
```

i think the documentation needs to be updated for 5.x for this stuff...

**Comment by baptisteArno** [comment]
[Here is a reproduction](https://codesandbox.io/s/jsplumb-1074-4bvyh)

I might be missing something? Also, the `filter` param doesn't seem to be in the `DragOptions` type.

**Comment by sporritt** [comment]
In this example the element jsPlumb is managing is the one that has `.no-drag`:

<img width="363" alt="image" src="https://user-images.githubusercontent.com/262720/143541645-c02c12f9-4206-4283-8d29-2eed9777191a.png">

the drag filter selector is meant to identify child elements of some draggable element that should not kick off a drag. for example, it might be weird if a user could drag a node around by a button that's on it.

If what you want is for a specific type of element to not be draggable then you need to set a `data-jtk-not-draggable` attribute on it:

<img width="353" alt="image" src="https://user-images.githubusercontent.com/262720/143542005-a8b65843-9e7e-4aca-bd7f-2d1c631e5a58.png">


it doesnt need a value.  you can toggle that attribute to enable/disable.

**Comment by baptisteArno** [comment]
Thank you for your fast reply, that's really appreciated 🙏

Oh ok, I just wanted to simplify the reproduction but you can set `no-drag` class inside the `StepNode` element, it won't prevent the drag either.

I've updated the sandbox accordingly

**Comment by sporritt** [comment]
So it seems the way to do it is via a method called `addDragFilter`

```
useEffect(() => {
    if (!graphRef.current) return;
    const jsPlumbInstance = newInstance({
      container: graphRef.current
    });

    jsPlumbInstance.addDragFilter(".no-drag, .no-drag *");

    setPlumbInstance(jsPlumbInstance);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```

Note the selector targets both `.no-drag` and also any child elements of it.

**Comment by baptisteArno** [comment]
This seems to work 💪

Let me know if I can contribute in any way

**Comment by sporritt** [comment]
i'm actually going to reopen this because it's a bit voodoo.  it _should_ work the way i told you it worked up there!

**Comment by sporritt** [comment]
I've just released 5.2.0 and you can do this now:

```
jsPlumbBrowserUI.newInstance({
  dragOptions:{
    filter:".no-drag"
 }
});
```

internally it just calls that `addDragFilter` method.
